File: employee_management_app/lib/models/employee.dart
class Employee {
  final int id;
  final String name;
  final String email;
  final String position;
  final int departmentId;
  final String departmentName;
  final double salary;
  final String phoneNumber;
  final String address;
  final String joiningDate;
  final String? profileImage;

  Employee({
    required this.id,
    required this.name,
    required this.email,
    required this.position,
    required this.departmentId,
    required this.departmentName,
    required this.salary,
    required this.phoneNumber,
    required this.address,
    required this.joiningDate,
    this.profileImage,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      position: json['position'],
      departmentId: json['department_id'],
      departmentName: json['department']['name'],
      salary: double.parse(json['salary'].toString()),
      phoneNumber: json['phone_number'],
      address: json['address'],
      joiningDate: json['joining_date'],
      profileImage: json['profile_image'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'email': email,
      'position': position,
      'department_id': departmentId,
      'salary': salary,
      'phone_number': phoneNumber,
      'address': address,
      'joining_date': joiningDate,
    };
  }
}

File: Pengumpulan/Program/Laravel/app/Models/Employee.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Employee extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'email',
        'position',
        'department_id',
        'salary',
        'phone_number',
        'address',
        'joining_date',
        'profile_image',
    ];

    protected $casts = [
        'salary' => 'decimal:2',
        'joining_date' => 'date',
    ];

    public function department(): BelongsTo
    {
        return $this->belongsTo(Department::class);
    }

    public function attendances(): HasMany
    {
        return $this->hasMany(Attendance::class);
    }
}
